src/main/java/org/apache/mesos/hdfs/Scheduler.java

```diff
@@ -416,7 +418,7 @@ private boolean tryToLaunchJournalNode(SchedulerDriver driver, Offer offer) {
     log.info(deadJournalNodes);
 
     if (deadJournalNodes.isEmpty()) {
-      if (liveState.getJournalNodeSize() == conf.getJournalNodeCount()) {
+      if (persistentState.getJournalNodes().size() == conf.getJournalNodeCount()) {
```

Comment:
this contacts the zkState every time. What is the purpose of using persistent state over live state?